### PR TITLE
Cumulative PR: Email Queuing Support + Shipment Issue Fix + store_id on item-level var

### PR DESCRIPTION
**Pull request by giraclion**
- Email Queuing Support
- Added Store ID as item-level var in purchases
- FIx for shipping Issue https://github.com/sailthru/sailthru-magento2-extension/issues/65

### CODE REVIEW

File: Mail/Queue/EmailSendPublisher.php [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+     *
+     * @param array $messageData
+     * @param int $attempt
+     *
+     * @return $this
+     */
+    public function execute(array $messageData, int $attempt = 0)
+    {
+        if ($attempt >= SettingsHelper::QUEUE_ATTEMPTS_COUNT) {
+            return $this;
```
Comment (brownac) [inline Mail/Queue/EmailSendPublisher.php]: Is there a way we can dead-letter send attempts somewhere if they pass the threshold? I just worry about the ability to send via Sailthru being affected in someway and losing a bunch of queue'd up emails.

File: Mail/Queue/EmailSendProcessor.php [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    {
+        $decodedData = json_decode($messageData, true);
+        try {
+            $this->getTransport($decodedData)->sendMessage();
+        } catch (\Throwable $t) {
+            $this->logger->critical($t->getMessage());
+            $this->publisher->execute($decodedData, ($decodedData['attempt'] ?? 0) + 1);
+        }
+
+        return $this;
```
Comment (brownac) [inline Mail/Queue/EmailSendProcessor.php]: Does this function need to return anything? Can it just be void?

File: Mail/Queue/EmailSendPublisher.php [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        }
+        $messageData['attempt'] = $attempt;
+        $this->publisher->publish($this->getTopicName(), json_encode($messageData));
+
+        return $this;
```
Comment (brownac) [inline Mail/Queue/EmailSendPublisher.php]: Does this function need to return anything? Can it just be void?

**Comment by brownac** [comment]
I am going to change the destination branch so that I can pull it down and test it locally.

File: Mail/Queue/EmailSendProcessor.php (hunk 2 above)
Comment (giraclion) [reply Mail/Queue/EmailSendProcessor.php]: This function can be void, but it was chosen to always return at least current object - this is matter of coding standard choice. Should I keep void?

File: Mail/Queue/EmailSendPublisher.php (hunk 3 above)
Comment (giraclion) [reply Mail/Queue/EmailSendPublisher.php]: This function can be void, but it was chosen to always return at least current object - this is matter of coding standard choice. Should I keep void?

**Pull request merged**